lua/dap.lua

```diff
@@ -1186,11 +1187,43 @@ function Session:_step(step)
   local thread_id = self.stopped_thread_id
   self.stopped_thread_id = nil
   vim.fn.sign_unplace(ns_pos)
-  session:request(step, { threadId = thread_id; }, function(err0, _)
-    if err0 then
-      print('Error on '.. step .. ': ' .. err0.message)
-    end
-  end)
+
+  local function send_request(target_id)
+    self:request(step, { threadId = thread_id; targetId = target_id; }, function(err0, _)
+      if err0 then
+        print('Error on '.. step .. ': ' .. err0.message)
+      end
+    end)
+  end
+
+  if self.capabilities.supportsStepInTargetsRequest
+    and M.defaults[self.config.type].ask_step_in_targets
+    and self.current_frame then
+    self:request("stepInTargets", { frameId = self.current_frame.id }, function(err1, response)
+      if err1 then
+        print('Error on '.. step .. ': ' .. err1.message .. " (while requesting stepInTargets)")
+      else
+        if #response.targets > 1 then
```

Comment:
should the UI change to just select the one entry if there is only one

Reply:
I think it makes sense to just step into without prompt if there is only 1 target. So would keep it as it is.

Reply:
I meant whether we should generalize this to all `ui.pick` decisions

Reply:
Oh. I'm not sure. `select_config_and_run` also behaves like that - so I guess it would make sense.

Would you change the default `ui.pick_one` implementation? It's intended to be overriden with fancier solutions, so we have to keep that in mind.
Might make sense to make it more explicit and have a separate method around `ui.pick_one`